server/internal/infrastructure/memory/project.go

```diff
@@ -0,0 +1,137 @@
+package memory
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase"
+	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/project"
+	"github.com/reearth/reearth-cms/server/pkg/rerror"
+)
+
+type Project struct {
+	lock sync.Mutex
+	data map[id.ProjectID]*project.Project
+	f    repo.WorkspaceFilter
+}
+
+func NewProject() repo.Project {
+	return &Project{
+		data: map[id.ProjectID]*project.Project{},
+	}
+}
+
+func (r *Project) Filtered(f repo.WorkspaceFilter) repo.Project {
+	return &Project{
+		// note data is shared between the source repo and mutex cannot work well
```

Comment:
```suggestion
```

This comment is no longer right because util.SyncMap is used and it's safe to be referred from various threads at the same time